libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -843,8 +843,18 @@ export class BotFrameworkAdapter extends BotAdapter implements ConnectorClientBu
             context.turnState.set(BotCallbackHandlerKey, logic);
             await this.runMiddleware(context, logic);
 
-            // Retrieve cached invoke response.
-            if (request.type === ActivityTypes.Invoke) {
```

Comment:
and this change presumably for consistency with the other place this condition occurs...

Reply:
Yes, these changes are ported from C# with exactly the same spirit (and tests) in mind. That said, I will add some more complex unit tests tomorrow to cover the interactions between the BFAdapter and ActivityHandler, and the two with SkillDialog.